### DISCUSSION TITLE
Generating CSS modules for React components in Bit

### DISCUSSION THREAD

**Discussion by alexf2** [teambit/bit#5658] [Help] [answered]
We started sharing a number of React components in our SPA as Bit.dev components and ran into an issue: can't use our components on Bit's dashboard due to CSS-modules as Bit's build process does not create them. We use Bit 0.0.687. Use "bit start" to launch Bit dashboard or "bit export" to publish the components to a remote scope, then open a remote dashboard. Our components which use: import style from './style.css' and relay on a CSS module get undefined "style" under Bit. May someone tell, please, if there is a way to alter Bit's build process to generate CSS modules? In our Application's Webpack build we use:
```module: {
  rules: [
    {
      test: /\.css$/i,
      include: /src/,
      exclude: /node_modules/,
      use: [
        {
          loader: require.resolve('style-loader', {
            paths: [require.resolve('webpack-config-single-spa')],
          }),
        },
        {
          loader: require.resolve('css-loader', {
            paths: [require.resolve('webpack-config-single-spa')],
          }),
          options: {
            importLoaders: 1,
            sourceMap: true,
            modules: {
              localIdentName: '[name]__[local]___[hash:base64:5]',
            },
          },
        },
        {
          loader: 'postcss-loader',
          options: {
            sourceMap: true,
          },
        },
      ],
    }
  ]}
```
and Webpack 5.x.

**Comment by itaymendel** [comment]
You can add another build-step to do post-processing for component's SCSS stuff (see [here](https://bit.dev/docs/dev-services/builder/implement-build-task) about build tasks).

There may be negative side-effects for this, as a consuming bundler may have CSS in the pre-built assets that could have been deduped...

**Reply by alexf2** [reply]
Thanks, itaymendel!

We do not use SCSS or LESS, only raw CSS.
Do you mean: we could add a build task, which compiles CSS against an additional Webpack loaders configuration, which I provided above, then dedublicate the output?
I suspect, Bit uses CRA under the hood. As I can see here (https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/), CRA automatically builds CSS-modules if a file has ".module.css" in it's name.
So, there is possible solution, which could help: "to rename all the CSS-files in our project according that pattern". Any other idea?

**Comment by alexf2** [comment, marked as answer]
So, renaming CSS files to style.module.css solved the issue.
